src/XMakeTasks/Copy.cs

```diff
@@ -408,6 +409,20 @@ CopyFileWithState copyFile
                 _copiedFiles = new TaskItem[0];
                 return true;
             }
+            
+            if (_destinationFiles != null)
+            {
+                var sameDestinationFiles = _destinationFiles.GroupBy(x => x).Where(g => g.Count() > 1).Select(y => y.First());
```

Comment:
Hmm, although localization might make the multiple-inputs-in-one-message thing tricky. Maybe a "related" message for the second + subsequent sources?